src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs

```diff
@@ -23,7 +24,7 @@ internal static class ProjectPackagesPrintUtility
         /// <param name="transitive">Whether include-transitive flag exists or not</param>
         /// <param name="outdated">Whether outdated flag exists or not</param>
         /// <param name="deprecated">Whether deprecated flag exists or not</param>
-        internal static PrintPackagesResult PrintPackages(
+        internal static async Task<PrintPackagesResult> PrintPackages(
```

Comment:
I think this is a code smell. `PrintPackages` writes stuff to the console, which is not async. Just as in the font end you create view models (which can be async) and then rendering the view is not async, `PrintPackages` is the equivalent to rendering a view.

Can the async work be moved to happen before calling `PrintPackages`, so that `PrintPackages` does nothing more than output stuff in the model to the console?

Reply:
I agree, but I was wary of making major structural changes to the client code given that I'm not familiar with it or its tests. I will look into it and see if there's something simple I can do, but I don't think making this method asynchronous is that big of a deal.

Reply:
I looked into it some more. The core problem here is that we're depending on a model that itself contains `async` properties that we care about, so, in order to render the model, we need to access the model's `async` properties.

In that case, if the goal is remove all `async` calls from this method, I see two options for resolving this:
1. Make a new model that contains a synchronously accessible deprecation metadata and convert our existing model to it before passing to `ProjectPackagesPrintUtility`.
2. Make the Package Manager UI use its own model with an asynchronously accessible deprecation metadata and revert the changes to all the other files here.

I would rather not go with the first option because it would essentially bloat this code path exclusively--regardless of whether or not we want to call the `async` properties in `ProjectPackagesPrintUtility` or in `ListPackageCommandRunner`, adding a new model between the two just adds to the complexity of the code, and I don't think removing the `async` warrants this.

The reason I decided against the second option while writing this code initially is that `IPackageSearchMetadata` already has an `async` property for fetching all the metadata for the other versions of the package, and I felt continuing that pattern would be the easiest change to move forward with. In general, the code didn't change much as a result of doing this, and the method that was most affected was this one.